src/components/EditInputGroup.vue

```diff
@@ -19,42 +19,115 @@
+            if (index === -1) {
+                index = this.$parent.recipe[this.fieldName].length
+            }
+            this.$parent.addEntry(this.fieldName, index, content)
+
+            if (focusAfterInsert) {
+                let $ul = $(this.$refs['list'])
+                let $this = this
+                this.$nextTick(function() {
+                    if ($ul.children('li').length > index) {
+                        if ($this.fieldType === 'text') {
+                            $ul.children('li').eq(index).find('input').focus()
+                        } else if ($this.fieldType === 'textarea') {
+                            $ul.children('li').eq(index).find('textarea').focus()
+                        }
                     }
-                    window.clearInterval(focusMonitor)
-                }
-                failSafe -= 100
-                if (!failSafe) {
-                    window.clearInterval(focusMonitor)
-                }
-            }, 100)
+                })
+            }
         },
         /**
          * Delete an entry from the list
          */
         deleteEntry: function(idx) {
             this.$parent.deleteEntry(this.fieldName, idx)
         },
+        /** 
+         * Handle typing in input or field or textarea
+         */
+        inputFieldChange: function(e) {  
+            // wait a tick to check if content was typed or pasted
+            this.$nextTick(function() {
+                if (this.contentPasted) {
+                    this.contentPasted = false
+                    return
+                }
+                let $li = $(e.currentTarget).parents('li')
+                this.$parent.recipe[this.fieldName][$li.index()] = e.target.value
+            })
+        },
+        /** 
+         * Handle paste in input field or textarea
+         */
+        handlePaste: function(e) {
+            this.contentPasted = true
+            if (!this.createFieldsOnNewlines) {
+                return
+            }
+
+            // get data from clipboard to keep newline characters, which are stripped
+            // from the data pasted in the input field (e.target.value)
+            var clipboardData = e.clipboardData || window.clipboardData;
+            var pastedData = clipboardData.getData('Text');
+
+            let input_lines_array = pastedData.split(/\r\n|\r|\n/g)
+            if ( input_lines_array.length == 1) {
+                return
+            }
+            e.preventDefault();
+
+            let $li = $(e.currentTarget).parents('li')
+            let $inserted_index = $li.index()
+            let $ul = $li.parents('ul')
+
+            if (this.fieldType === 'textarea') {
+                for (let i = input_lines_array.length-1; i >= 0; --i)
+                {
+                    if (input_lines_array[i].trim() == '') {
+                        input_lines_array.splice(i, 1)
+                    }
+                }
+            }
```

Comment:
Should this check apply to ingredients and tools as well? Or is there some particular reason why empty lines should be preserved?

Reply:
I think the idea is to have empty lines for paragraphs in the instructions if needed.

Reply:
Although this could be related to #381

Reply:
Yeah, I don’t know. My reasoning was that currently we have the possibility/workaround to separate ingredient groups by using `##`. So if there is an empty line, the user could insert a caption. While grouping instructions might have a use case as well, there is currently no nice way to reflect this in the UI.

Reply:
I get the idea (and it's not a bad one). I just have a concern that when copy-pasting lists of ingredients from web pages there could be empty lines in between just because there are some imperceptible HTML elements embedded into the list and the the user would have to manually remove one or two or more empty lines for every ingredient/tool that was added. Maybe adding the odd caption between pasting the lists of ingredients or even afterwards would be the lesser evil, or what do you think?

Reply:
You have a point there. What do you think could be those “imperceptible HTML elements”? If I have a list, I can clean the lines to check against an empty string.

Reply:
Well, a site that I frequently use wraps ingredients in <p></p> tags. When I copy-paste a list of ingredients from that site, I get an empty line between every ingredient. So I would have to manually remove all those empty lines whenever I used that site. Just removing the check for `fieldType==='textarea'` would automatically remove all those empty lines from my ingredient lists.

Reply:
Yes, true, I was thinking of some invisible special characters. Empty lines, or essentially everything that is removed by a `trim()` is easy :)